File: Pertemuan 3/RekeningBank.php
<?php
declare(strict_types=1);

class RekeningBank
{
    public const BUNGA_TAHUNAN = 0.025;
    public const BIAYA_ADMIN = 5000.0;
    public const BATAS_TARIK_SEKALI = 5_000_000.0;

    private static int $jumlahRekening = 0;

    private float $saldo;

    public function __construct(
        private readonly string $nomor,
        private readonly string $pemilik,
        float $saldoAwal = 0,
    ) {
        if (trim($this->nomor) === '') {
            throw new InvalidArgumentException('Nomor rekening tidak boleh kosong');
        }
        // is_finite menolak NAN dan INF; sisanya menolak negatif
        if (!is_finite($saldoAwal) || $saldoAwal < 0) {
            throw new InvalidArgumentException('Saldo awal harus angka valid dan tidak negatif');
        }

        $this->saldo = $saldoAwal;
        self::$jumlahRekening++; // hanya tercapai jika semua validasi lolos
    }

    public static function rekeningPelajar(string $nomor, string $pemilik): static
    {
        return new static($nomor, $pemilik, 0);
    }

    public function setor(float $jumlah): void
    {
        if (!is_finite($jumlah) || !($jumlah > 0)) {
            throw new InvalidArgumentException('Jumlah setoran harus angka valid lebih dari 0');
        }
        $this->saldo += $jumlah;
    }

    public function tarik(float $jumlah): void
    {
        if (!is_finite($jumlah) || !($jumlah > 0)) {
            throw new InvalidArgumentException('Jumlah penarikan harus angka valid lebih dari 0');
        }
        if ($jumlah > self::BATAS_TARIK_SEKALI) {
            throw new InvalidArgumentException(
                'Penarikan melebihi batas sekali transaksi (Rp'
                . number_format(self::BATAS_TARIK_SEKALI, 0, ',', '.') . ')'
            );
        }
        if ($jumlah > $this->saldo) {
            throw new RuntimeException('Saldo tidak mencukupi');
        }
        $this->saldo -= $jumlah;
    }

    public function potongBiayaAdmin(): void
    {
        $this->saldo = max(0.0, $this->saldo - self::BIAYA_ADMIN);
    }

    public static function getJumlahRekening(): int
    {
        return self::$jumlahRekening;
    }

    public static function bungaSetahun(float $pokok): float
    {
        return $pokok * self::BUNGA_TAHUNAN;
    }

    public function getSaldo(): float { return $this->saldo; }
    public function getNomor(): string { return $this->nomor; }

    public function __toString(): string
    {
        return sprintf('Rekening[%s] %-14s Rp%s',
            $this->nomor, $this->pemilik, number_format($this->saldo, 2, ',', '.'));
    }
}

File: Pertemuan 3/RekeningBank.java
public class RekeningBank {

    public static final double BUNGA_TAHUNAN = 0.025;
    public static final double BIAYA_ADMIN = 5000;
    public static final double BATAS_TARIK_SEKALI = 5_000_000;

    private static int jumlahRekening = 0;

    private final String nomor;
    private final String pemilik;
    private double saldo;

    public RekeningBank(String nomor, String pemilik) {
        this(nomor, pemilik, 0);
    }

    public RekeningBank(String nomor, String pemilik, double saldoAwal) {
        if (nomor == null || nomor.isBlank()) {
            throw new IllegalArgumentException("Nomor rekening tidak boleh kosong");
        }
        // !(x >= 0) menolak negatif DAN NaN; isInfinite menolak tak hingga
        if (!(saldoAwal >= 0) || Double.isInfinite(saldoAwal)) {
            throw new IllegalArgumentException("Saldo awal harus angka valid dan tidak negatif");
        }

        this.nomor = nomor;
        this.pemilik = pemilik;
        this.saldo = saldoAwal;

        jumlahRekening++; // hanya di sini; lolos validasi dulu baru dihitung
    }

    public void setor(double jumlah) {
        if (!(jumlah > 0) || Double.isInfinite(jumlah)) {
            throw new IllegalArgumentException("Jumlah setoran harus angka valid lebih dari 0");
        }
        this.saldo += jumlah;
    }

    public void tarik(double jumlah) {
        if (!(jumlah > 0) || Double.isInfinite(jumlah)) {
            throw new IllegalArgumentException("Jumlah penarikan harus angka valid lebih dari 0");
        }
        if (jumlah > BATAS_TARIK_SEKALI) {
            throw new IllegalArgumentException(
                    String.format("Penarikan melebihi batas sekali transaksi (Rp%,.0f)", BATAS_TARIK_SEKALI));
        }
        if (jumlah > saldo) {
            throw new IllegalArgumentException("Saldo tidak mencukupi");
        }
        this.saldo -= jumlah;
    }

    public void potongBiayaAdmin() {
        this.saldo = Math.max(0, this.saldo - BIAYA_ADMIN);
    }

    public static int getJumlahRekening() {
        return jumlahRekening;
    }

    public static double bungaSetahun(double pokok) {
        return pokok * BUNGA_TAHUNAN;
    }

    public double getSaldo()  { return saldo; }
    public String getNomor()  { return nomor; }

    @Override
    public String toString() {
        return String.format("Rekening[%s] %-14s Rp%,.2f", nomor, pemilik, saldo);
    }
}